themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js

```diff
@@ -935,17 +935,17 @@ module.controller('RealmIdentityProviderCtrl', function($scope, $filter, $upload
 
     $scope.initSamlProvider = function() {
         $scope.nameIdFormats = [
-            /*
-            {
-                format: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
-                name: "Transient"
-            },
-            */
+         
+           
```

Comment:
Remove the empty lines.
Also at the end of this `initSamlProvider` function, set the default value for `allowCreate` flag, something like:
```javascript
        $scope.identityProvider.config.allowCreate = 'true';
```
This should be `true` for the time being to stay backwards-compatible. Changing it to false may be done in the future.

Reply:
I set default value to false in order to be compatible with default SAML value (Section 3.4.1.1 in https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf). However, I will change it as you desired.

Reply:
You are indeed right about the specification. The reason for defaulting to `true` is to stay conformant to the current behaviour. In the future, the default might be changed, and that would impact also the default that is set here:
https://github.com/keycloak/keycloak/blob/809ab5bf002f0f34781e27be7cbfb29bcef793bd/services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java#L134-L135

Reply:
Unresolving this conversation as this comment needs to be addressed at the proper place